src/components/password/password.spec.tsx

```diff
@@ -0,0 +1,239 @@
+      input = wrapper.find("input").at(0);
+      const firstFocusableElement = document.querySelector("input");
+      expect(document.activeElement).not.toBe(firstFocusableElement);
+    });
+  });
+
+  describe("password prop checks", () => {
+    beforeAll(() => {
+      wrapper = renderPasswordInput({});
+    });
+
+    it("default input type is `password`", () => {
+      expect(wrapper.find("input").at(0).prop("type")).toBe("password");
+    });
+
+    it("input type should change from `password` to `text` on click", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      wrapper.update();
+      expect(wrapper.find("input").at(0).prop("type")).toBe("text");
+    });
+
+    it("autoComplete attribute is `off`", () => {
+      expect(wrapper.find("input").at(0).prop("autoComplete")).toBe("off");
+    });
+
+    it("input id should equal value of `id` prop when passed", () => {
+      wrapper = renderPasswordInput({ id: "foo" });
+      expect(wrapper.find("input").at(0).prop("id")).toBe("foo");
+    });
+
+    it("input id should be a generated guid when `id` prop is undefined", () => {
+      wrapper = renderPasswordInput({ id: "" });
+      expect(wrapper.find("input").at(0).prop("id")).toBe(mockedGuid);
+    });
+  });
+
+  describe("buttonMinor prop checks", () => {
+    it("aria-controls attribute is correct", () => {
+      wrapper = renderPasswordInput({ id: "foo" });
+      expect(wrapper.find(StyledButtonMinor).prop("aria-controls")).toBe("foo");
+    });
+
+    it("buttonType attribute is correct", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("buttonType")).toBe(
+        "tertiary"
+      );
+    });
+
+    it("iconPosition attribute is correct", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("iconPosition")).toBe(
+        "before"
+      );
+    });
+
+    it("size attribute is correct", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("size")).toBe("small");
+    });
+
+    it("default label text should be 'Show`", () => {
+      expect(wrapper.find(StyledButtonMinor).text()).toBe("Show");
+    });
+
+    it("label text should change from `Show` to `Hide` onClick", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      expect(wrapper.find(StyledButtonMinor).text()).toBe("Hide");
+    });
+
+    it("default aria-label text should be 'Show password`", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("aria-label")).toBe(
+        "Show password"
+      );
+    });
+
+    it("aria-label text should change from `Show Password` to `Hide password` onClick", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      expect(wrapper.find(StyledButtonMinor).prop("aria-label")).toBe(
+        "Hide password"
+      );
+    });
+
+    it("default iconType should be 'view'", () => {
```

Comment:
You've used a mixture of back ticks and single quotes. Like here and line 158 for example, best to pick one and stick to that for consistency.